sys/console/full/src/console.c

```diff
@@ -168,11 +170,24 @@ console_write(const char *str, int cnt)
     if (console_lock(timeout) != OS_OK) {
         return;
     }
+        
+    /* If the byte string is non nlip and we are silencing non nlip bytes,
+     * do not let it go out on the console
+     */ 
+    if (!g_is_output_nlip && g_console_silence_non_nlip) {
+        goto done;
+    }
+ 
     for (i = 0; i < cnt; i++) {
         if (console_out_nolock((int)str[i]) == EOF) {
             break;
         }
     }
+
+done:
+    if (str[cnt - 1] == '\n') {
```

Comment:
Someone could conceivable pass a `cnt` value of 0.  It might be a good idea to make sure `cnt` is greater than 0 before subtracting one.